### PR TITLE
Java Bytecode parsing tidy

### PR DESCRIPTION
**Pull request by thk123**
While investigating a bug, tried to make a few things a little clearer.

### CODE REVIEW

File: src/java_bytecode/java_bytecode_convert_method.cpp [hunk 1]
```diff
@@ -45,6 +45,9 @@ Author: Daniel Kroening, kroening@kroening.com
 #include <unordered_set>
 #include <regex>
 
+/// Given a string of the format '?blah?', will return true when compared
+/// against a string that matches appart from any characters that are '?'
+/// in the original string. Equivalent to doing a regex match on '.blah.'
```
Comment (thk123) [inline src/java_bytecode/java_bytecode_convert_method.cpp:50]: If it doesn't take to long I intend to remove this completely and replace its uses with `std::regex` unless anyone objects to that?

File: src/java_bytecode/java_bytecode_parser.cpp [hunk 2]
```diff
@@ -946,6 +947,12 @@ void java_bytecode_parsert::rmethod_attribute(methodt &method)
     {
       u2 start_pc=read_u2();
       u2 end_pc=read_u2();
+
+      INVARIANT(
```
Comment (mgudemann) [inline src/java_bytecode/java_bytecode_parser.cpp:951]: would you mind adding that this invariant is from the class file format spec ("4.7.3. The Code Attribute" for Java8)

File: src/java_bytecode/java_bytecode_parser.cpp [hunk 3]
```diff
@@ -79,6 +79,7 @@ class java_bytecode_parsert:public parsert
     if(index==0 || index>=constant_pool.size())
     {
       error() << "invalid constant pool index (" << index << ")" << eom;
+      error() << "constant pool size: " << constant_pool.size() << eom;
```
Comment (mgudemann) [inline src/java_bytecode/java_bytecode_parser.cpp:82]: I think it makes still sense to output the accessed `index`, too

File: src/java_bytecode/java_bytecode_convert_method.cpp (hunk 1 above)
Comment (smowton) [reply src/java_bytecode/java_bytecode_convert_method.cpp:50]: Knock up a quick standalone benchmark before you do, as this is obviously a much simpler machine

**Review by smowton — APPROVED** [review]

File: src/java_bytecode/java_bytecode_parser.cpp (hunk 3 above)
Comment (thk123) [reply src/java_bytecode/java_bytecode_parser.cpp:82]: This is still done on the line above.

**Pull request merged**